script: Clean up `pulsar` and `ppm` on uninstall

DeeDeeG opened the pull request:
For Linux .rpm and .deb targets only.

<details><summary><b>Requirements for Contributing a Bug Fix</b> (from template, click to expand):</summary>

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix an existing bug. To contribute other changes, you must use a different template. You can see all templates at <https://github.com/atom/.github/tree/master/.github/PULL_REQUEST_TEMPLATE>.
* The pull request must update the test suite to demonstrate the changed functionality. For guidance, please see <https://flight-manual.atom.io/hacking-atom/sections/writing-specs/>.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see <https://github.com/atom/.github/tree/master/CONTRIBUTING.md#pull-requests>.

</details>

### Identify the Bug

<!--

Link to the issue describing the bug that you're fixing.

If there is not yet an issue for your bug, please open a new issue and then link to that issue in your pull request.
Note: In some cases, one person's "bug" is another person's "feature." If the pull request does not address an existing issue with the "bug" label, the maintainers have the final say on whether the current behavior is a bug.

-->

Follow-up to:
- https://github.com/pulsar-edit/pulsar/pull/82
- https://github.com/pulsar-edit/pulsar/pull/228
- https://github.com/pulsar-edit/pulsar/pull/273

### Description of the Change

<!--

We must be able to understand the design of your change from this description. If we can't get a good idea of what the code will be doing from the description here, the pull request may be closed at the maintainers' discretion. Keep in mind that the maintainer reviewing this PR may not be familiar with or have worked with the code here recently, so please walk us through the concepts.

-->

When uninstalling the .`deb` and `.rpm` packages on Linux, clean up the `pulsar` (`pulsar.sh`) and `ppm` (symlink to `/opt/Pulsar/resources/app/ppm/bin/apm`) stuff from `/usr/bin`.

(Note: These are only automatically created for the `.deb` and `.rpm` Linux packages. So there is nothing to clean up on other OSes or targets, other than the tangentially related but off-topic for this PR macOS bins users have to manually request installation of.)

**Relevant docs:** https://www.electron.build/configuration/linux#linuxtargetspecificoptions-apk-freebsd-pacman-p5p-and-rpm-options

### Alternate Designs

<!-- Explain what other alternates were considered and why the proposed version was selected -->

Not cleaning up is bad, so I didn't consider just not cleaning up. However:

_I do want_ to move these from **`/usr/bin`** (distro packages install here and there are obscure conflicts with a `ppm` bin from a Perl package manager package, and with a bin `pulsar` from an MRI science package called `odin`) to **`/usr/local/bin`**, which is coincidentally where Atom installed its `atom` and `apm` bins to, presumably for the same reason.

(See: https://pkgs.org/search/?q=%2Fusr%2Fbin%2Fpulsar and https://pkgs.org/search/?q=%2Fusr%2Fbin%2Fppm)

**BUT*:* IMO we need to have this cleanup code out in the wild for a while to clean up all the old bins people have installed, before making the move to `/usr/local/bin`.

### Possible Drawbacks

<!-- What are the possible side-effects or negative impacts of the code change? -->

I would say there is a risk of deleting the Perl package manager `ppm` or the MRI science `pulsar` bin from Odin... but hey, we have already been clobbering these with our own bins! So... It's already too late for that! Oops. No drawbacks that I can think of that would be new as of this PR.

So yeah, let's move these to `/usr/local/bin` instead of `/usr/bin`, after about a month or so, say?

### Verification Process

<!--

What process did you follow to verify that the change has not introduced any regressions? Describe the actions you performed (including buttons you clicked, text you typed, commands you ran, etc.), and describe the results you observed.

-->

**TODO:** Grab one of the built packages from Cirrus and test this out. I'll be testing the .deb package, personally.

### Release Notes

<!--

Please describe the changes in a single line that explains this improvement in
terms that a user can understand. This text will be used in Atom's release notes.

If this change is not user-facing or notable enough to be included in release notes
you may use the strings "Not applicable" or "N/A" here.

Examples:

- The GitHub package now allows you to add co-authors to commits.
- Fixed an issue where multiple cursors did not work in a file with a single line.
- Increased the performance of searching and replacing across a whole project.

-->

Clean up `pulsar` and `ppm` bins/scripts/symlinks upon uninstalling the `.rpm` or `.deb` packages.

confused-Techie reviewed (approved):
The code looks solid, but I'm not in a position to quite test. But seeing @mauricioszabo's approval is encouraging, so we should be good to go

DeeDeeG commented:
**Update:** I tested it, and this works. :+1:

---

Something related I noticed: I was using `test -L` ("is the path a symlink?") for ppm's symlink on the post-uninstall script, but `test -f` ("is the path a regular file?") on the post-install script. `-L` is more correct. (See the commit message for much, much verbose explanation of this.) So **I added a commit here that slightly improved the correctness of the `post-install.sh` script, by switching from `[ -f ]` to `[ -L ]` in the `post-install.sh` script.**

Still on the same theme of cleaning up correctly, sort of? So I added it to this same PR.

Would appreciate another review, but it's such a tiny change, and I tested it pretty thoroughly on my machine, so I may merge if no-one gets around to reviewing.

confused-Techie reviewed (approved):
My approval still applys that the code looks solid, and if you were able to test and ensure this works that's awesome.

Would you like me to post the binary this creates on Discord so we can try to have others tests? Otherwise if you feel your system is a good example for what's expected here lets get it merged

Daeraxa commented:
Just had a try on MX Linux and yup, `pulsar` and `ppm` are created but after installing `which pulsar` and `which ppm` return nothing and `/opt/Pulsar` no longer exists.

DeeDeeG commented:
@confused-Techie yeah, I'm confident `-L` works, as seen in https://github.com/pulsar-edit/pulsar/pull/273, and as tested again here. And other than that bit, me and Spiker985 discussed the basic code pattern for these shell scripts in https://github.com/pulsar-edit/pulsar/pull/228, and I feel pretty comfortable with what we came up with. People are welcome to improve it if there's a reason for it, but it's pretty simple and it's working, so I am comfortable with it.

Was away from the project for the weekend, thanks for the approves @mauricioszabo and @confused-Techie, thanks for testing and confirming it works @Daeraxa. Merging this now!!
